Lib/unittest/suite.py

```diff
@@ -264,12 +264,12 @@ def _handleModuleTearDown(self, result):
                                                         previousModule)
             finally:
                 _call_if_exists(result, '_restoreStdout')
-                try:
```

Comment:
Should it *also* be called in this finally block? That's the previous behavior, and I imagine it could be important to ensure it's called even when tearDownModule() raises a BaseException.

Reply:
Thanks for the suggestion. I think this is not need TBH, my understanding it that the finally block will be executed anyways in case of a Exception and, since `_call_if_exists` doesn't return, the clean up is done in the following steps.

Regarding the `BaseException`, yes, if `tearDownModule` raises it no doModuleCleanups is performed. 

I think it's a good idea to catch it but for consistency we should change `setUpClass` as well. What do you think?

        setUpClass = getattr(currentClass, 'setUpClass', None)
        if setUpClass is not None:
            _call_if_exists(result, '_setupStdout')
            try:
                setUpClass()
            except Exception as e:
                if isinstance(result, _DebugResult):
                    raise`